src/tests/meson.build

```
@@ -0,0 +1,310 @@
+libtest = static_library('common',
+    include_directories: core_inc,
+    sources: files('sc-test.c'),
+    link_with: [
+        libcompat,
+        libopensc_static
+    ]
+)
+
+tests_targets = []
+if host_machine.system() == 'windows' or host_machine.system() == 'cygwin'
+    tests_targets += modwindows.compile_resources(
+        configure_file(
+            configuration: conf_aux,
+            input: 'tests.rc.in',
+            output: 'tests.rc'
+        ),
+        depend_files: 'tests.manifest'
+    )
+endif
+
+
+executable('base64',
+    tests_targets,
+    include_directories: core_inc,
+    sources: files('base64.c'),
+    link_with: [
+        libopensc_static,
+        libtest
+    ]
+)
+
+executable('lottery',
+    tests_targets,
+    include_directories: core_inc,
+    sources: files('lottery.c'),
+    link_with: [
+        libopensc_static,
+        libtest
+    ]
+)
+
+executable('p15dump',
+    tests_targets,
+    include_directories: core_inc,
+    sources: files(
+        'p15dump.c',
+        'print.c'
+    ),
+    link_with: [
+        libopensc_static,
+        libtest
+    ]
+)
+
+executable('pintest',
+    tests_targets,
+    include_directories: core_inc,
+    sources: files(
+        'pintest.c',
+        'print.c'
+    ),
+    link_with: [
+        libcompat,
+        libopensc_static,
+        libtest
+    ]
+)
+
+executable('prngtest',
+    tests_targets,
+    include_directories: core_inc,
+    sources: files('prngtest.c'),
+    link_with: [
+        libopensc_static,
+        libtest
+    ]
+)
```

something like this?

maybe the extra files will need some more love to work correctly as this is untested and I am not much experienced with meson, but I find this much more readable and maintainable than the 50+ lines of expanded each of the binaries:

```suggestion
execs = {
  'base64': {},
  'lottery': {},
  'p15dump': {'print.c'},
  'pintest': {'print.c'},
  'prngtest': {},
};
foreach e, extra_files : execs
    executable(e,
      tests_targets,
      include_directories: core_inc,
      sources: files('@0@.c'.format(e)) + extra_files,
      link_with: [
          libopensc_static,
          libtest
      ]
  )
endforeach
```

I think very similarly could be handled unit tests below.